MIGRATION.ru.md

```diff
@@ -1,5 +1,743 @@
 # Миграция
 
+## 4.0.0
+
+### Отдельный блок `i-bem-dom`
+
+Элемент `dom` блока `i-bem` был перенесён в отдельный блок `i-bem-dom`.
+
+Было:
+
+```js
+modules.define('my-dom-block', ['i-bem__dom'], function(provide, BEMDOM) {
+    // ...
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', ['i-bem-dom'], function(provide, bemDom) {
+    // ...
+});
+```
+
+Блоки `i-bem` и `i-bem-dom` больше не являются классами, представляя собой модули с методами для декларации
+БЭМ-сущностей, ссылками на классы БЭМ-сущностей и некоторыми дополнительными хелперами. Эти методы больше не являются методами класса для соответсвующих блоков.
+
+Задача: [#413](https://github.com/bem/bem-core/issues/413).
+
+### Декларация
+
+### Декларация блока
+
+Для декларации блока, вместо метода `decl()`, следует использовать метод `declBlock()`.
+
+Было:
+
+```js
+modules.define('my-dom-block', ['i-bem__dom'], function(provide, BEMDOM) {
+
+provide(BEMDOM.decl(this.name, { /* ... */ }));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', ['i-bem-dom'], function(provide, bemDom) {
+
+provide(bemDom.declBlock(this.name, { /* ... */ }));
+
+});
+```
+
+### Декларация модификатора
+
+Для декларации модификатора, вместо метода `decl()`, следует использовать метод `declMod()`.
+
+Было:
+
+```js
+modules.define('my-dom-block', function(provide, MyDomBlock) {
+
+provide(MyDomBlock.decl({ modName : 'my-mod', modVal : 'my-val' }, { /* ... */ }));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', function(provide, MyDomBlock) {
+
+provide(MyDomBlock.declMod({ modName : 'my-mod', modVal : 'myVal' }, { /* ... */ }));
+
+});
+```
+
+### Доопределение блока
+
+Вместо метода `decl()` класса блока следует использовать метод `declBlock()` модуля `i-bem-dom`.
+
+Было:
+
+```js
+modules.define('my-dom-block', function(provide, MyDomBlock) {
+
+provide(MyDomBlock.decl({ /* ... */ }));
+
+});
+```
+
+Стало:
+
+```js
+modules.define('my-dom-block', ['i-bem-dom'], function(provide, bemDom, MyDomBlock) {
+
+provide(bemDom.declBlock(MyDomBlock, { /* ... */ }));
```

Comment:
For code
```js
modules.define('b1', ['i-bem-dom'], function(provide, bemDom) {

provide(bemDom.declBlock(this.name, {
    onSetMod: {
        js: {
            inited: function() {
                console.log('da');
            }
        }
    }
}));

});

modules.define('b1', ['i-bem-dom'], function(provide, bemDom, B1) {

provide(bemDom.declBlock(B1, {
    onSetMod: {
        js: {
            inited: function() {
                console.log('net');
            }
        }
    }
}));

});
```

I've got `da` in console.

But if I change `B1` with a string `'b1'` it works as expected. Is it a bug?